sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -2036,11 +2039,11 @@ export interface SearchAlias {
 }
 
 // @public
-export class SearchClient<T> implements IndexDocumentsClient<T> {
+export class SearchClient<T extends object> implements IndexDocumentsClient<T> {
     constructor(endpoint: string, indexName: string, credential: KeyCredential | TokenCredential, options?: SearchClientOptions);
     // @deprecated
     readonly apiVersion: string;
-    autocomplete<Fields extends keyof T>(searchText: string, suggesterName: string, options?: AutocompleteOptions<Fields>): Promise<AutocompleteResult>;
+    autocomplete<Fields extends SelectFields<T>>(searchText: string, suggesterName: string, options?: AutocompleteOptions<SearchPick<T, Fields>>): Promise<AutocompleteResult>;
```

Comment:
Does `getDocument` not need the same treatment?

Reply:
It does! It takes a `selectedFields` option. Thanks for catching that!